src/useLens.ts

```diff
@@ -0,0 +1,36 @@
+import { type DependencyList, useMemo } from 'react';
+import type { Control, FieldValues } from 'react-hook-form';
+
+import type { Lens } from './types/lenses';
+import { LensCore } from './LensCore';
+import { createLensesCache } from './utils';
+
+export interface UseLensProps<TFieldValues extends FieldValues = FieldValues> {
+  control: Control<TFieldValues>;
+}
+
+/**
+ * Creates lens from react-hook-form control.
+ *
+ * @example
+ * ```tsx
+ * function App() {
+ *   const { control } = useForm<{
+ *     firstName: string;
+ *   }>();
+ *
+ *   const lens = useLens({ control });
+ * }
+ * ```
+ */
+export function useLens<TFieldValues extends FieldValues = FieldValues>(
+  props: UseLensProps<TFieldValues>,
+  deps: DependencyList = [],
+): Lens<TFieldValues> {
+  return useMemo(() => {
+    const cache = createLensesCache();
```

Comment:
The cache is never reset because I haven't found a way to detect when a field control unsubscribes in order to remove the appropriate lens.

Reply:
Do you want me to expose a callback for input to be unmounted?

Reply:
I think yes, also there is an edge case when the same field is registered several time and then on of them is unregistered, in this case the lens cache should not be updated